pkg/volume/iscsi/iscsi_util.go

```diff
@@ -204,34 +204,63 @@ func (util *ISCSIUtil) MakeGlobalVDPDName(iscsi iscsiDisk) string {
 
 // persistISCSIFile saves iSCSI volume configuration for DetachDisk
 // into given directory.
-func (util *ISCSIUtil) persistISCSIFile(conf iscsiDisk, mnt string) error {
-	file := filepath.Join(mnt, "iscsi.json")
-	fp, err := os.Create(file)
+func (util *ISCSIUtil) persistISCSIFile(conf iscsiDisk, filePath string) error {
+	fp, err := os.Create(filePath)
 	if err != nil {
-		return fmt.Errorf("iscsi: create %s err %s", file, err)
+		return fmt.Errorf("iscsi: failed to create %s: %s", filePath, err)
 	}
 	defer fp.Close()
 	encoder := json.NewEncoder(fp)
 	if err = encoder.Encode(conf); err != nil {
-		return fmt.Errorf("iscsi: encode err: %v", err)
+		return fmt.Errorf("iscsi: failed to encode json: %s", err)
 	}
 	return nil
 }
 
 func (util *ISCSIUtil) loadISCSI(conf *iscsiDisk, mnt string) error {
-	file := filepath.Join(mnt, "iscsi.json")
-	fp, err := os.Open(file)
-	if err != nil {
-		return fmt.Errorf("iscsi: open %s err %s", file, err)
+	jsonFilePath := getISCSIPersistFile(mnt)
+	fp, openerr := os.Open(jsonFilePath)
+	if openerr != nil {
+		if !os.IsNotExist(openerr) {
+			return fmt.Errorf("iscsi: failed to open %s: %s", jsonFilePath, openerr)
+		}
+		// JSON file does not exist. For backward compatibility: try iscsi.json in the mount directory.
```

Comment:
For how long do we want to be backward-compatible with `iscsi.json`? Can we add a TODO to remove this v1.2x?

Reply:
TODO added